### PR TITLE
test: expand coverage with 31 new tests across 14 modules

### PR DESCRIPTION
**Pull request by axisrow**
## Summary
- Add 31 new tests covering argv composition, blank filter handling, batch limit validation, auth error scenarios, and empty result edge cases
- Tests added across 14 modules: agency, bidmodifiers, bids, businesses, clients, creatives, dynamic_ads, extensions, images, keyword_bids, leads, reports, research, turbo_pages
- Total test count: 385 → 416

## Test plan
- [x] All 416 tests pass, 8 skipped (unchanged)
- [x] No changes to production code — only test files modified

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by axisrow** [comment]
@claude review. Focus on critical issues: bugs, security vulnerabilities, logical errors, data loss risks, performance problems. Do NOT nitpick style, naming conventions, minor formatting, or subjective preferences — only flag issues that could break functionality or cause real harm in production.

File: tests/test_turbo_pages.py [hunk 1]
```diff
@@ -88,3 +88,45 @@ def test_turbo_pages_add_with_extra_json(self):
...
+                "--json",
+                '{"Description":"desc"}',
+            ]
+        )
+
+    def test_turbo_pages_list_empty_result(self):
+        """Test empty response returns empty dict."""
+        with patch(
+            "server.tools.turbo_pages.get_runner",
+            return_value=_mock_runner({"turboPages": []}),
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_turbo_pages.py:120]: SKIP: Test class grouping is cosmetic — these tests run correctly regardless of class placement. The review scope explicitly excluded style/organizational preferences.

File: tests/test_research.py [hunk 2]
```diff
@@ -57,3 +57,64 @@ def test_keywords_deduplicate_basic(self):
...
+            "server.tools.research.get_runner", return_value=_mock_runner(mock_result)
+        ):
+            result = keywords_deduplicate(keywords="keyword1,keyword2")
+            assert len(result["deduplicated"]) == 2
+
+    def test_keywords_has_volume_argv_composition(self):
+        """Test has_volume passes correct argv to CLI."""
+        runner = MagicMock()
+        runner.run_json.return_value = {"k1": True}
+        with patch("server.tools.research.get_runner", return_value=runner):
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_research.py:77]: SKIP: Test class grouping is cosmetic — these tests run correctly regardless of class placement. The review scope explicitly excluded style/organizational preferences.

File: tests/test_bids.py [hunk 3]
```diff
@@ -128,3 +128,34 @@ def test_bids_set_requires_update_fields(self):
...
+            assert result["success"] is True
+            call_args = runner.run_json.call_args[0][0]
+            assert "--bid" not in call_args
+            assert "--json" in call_args
+
+    def test_bids_list_ad_group_batch_limit(self):
+        """Test batch limit validation for ad_group_ids."""
+        ids = ",".join(str(i) for i in range(1, 12))  # 11 IDs
+        result = bids_list(ad_group_ids=ids)
+        assert "error" in result
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_bids.py:153]: SKIP: Test class grouping is cosmetic — these tests run correctly regardless of class placement. The review scope explicitly excluded style/organizational preferences.

File: tests/test_keyword_bids.py [hunk 4]
```diff
@@ -172,3 +172,32 @@ def test_keyword_bids_set_requires_changes(self):
...
+
+    def test_keyword_bids_list_ignores_blank_filters(self):
+        """Test blank filters behave like no filter."""
+        runner = MagicMock()
+        runner.run_json.return_value = SAMPLE_BIDS
+        with patch("server.tools.keyword_bids.get_runner", return_value=runner):
+            result = keyword_bids_list(
+                campaign_ids="   ", ad_group_ids="   ", keyword_ids="   "
+            )
+            assert len(result) == 1
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_keyword_bids.py:199]: SKIP: Test class grouping is cosmetic — these tests run correctly regardless of class placement. The review scope explicitly excluded style/organizational preferences.

File: tests/test_clients.py [hunk 5]
```diff
@@ -110,3 +110,30 @@ def test_update_client_with_grants(self):
...
+        )
+
+    def test_get_client_ignores_blank_ids(self):
+        """Test blank ids behave like no filter."""
+        runner = MagicMock()
+        runner.run_json.return_value = {"Clients": []}
+        with patch("server.tools.clients.get_runner", return_value=runner):
+            clients_get(ids="   ")
+            call_args = runner.run_json.call_args[0][0]
+            assert "--ids" not in call_args
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_clients.py:139]: SKIP: Test class grouping is cosmetic — these tests run correctly regardless of class placement. The review scope explicitly excluded style/organizational preferences.
